### ISSUE TITLE
Latest Firefox breaks the Carousel component on Bootstrap 4.1.3

### ISSUE THREAD

**Issue by neoschronos** [twbs/bootstrap#27525]
Updating to the latest version of Firefox breaks the Carousel component on Bootstrap 4.1.3

**Environment**
- OS: MacOS Sierra 10.12.6 (16G1510)
- Browser: Firefox Quantum 63.0 (64-bit) 

**Test Case for reproduction** 
- Visit https://getbootstrap.com/docs/4.1/components/carousel/
- See that carousels do not slide

**Notes**
This issue is posted here because previous Bootstrap versions do not show the problem, specifically

- No problem exists with Firefox 62.0.3 (64-bit)and Bootstrap 4.1.3
- No problem exists with Firefox 63.0 and Bootstrap 4.0.0

To verify Bootstrap 4.0.0. works visit https://getbootstrap.com/docs/4.0/components/carousel/ with the latest Firefox 63.0 (64-bit)

Cheers,
      Thomas

**Comment by Johann-S** [comment]
Carousel work on my FF 63 with Win10 but I don't have the slide animation 😨

**Comment by neoschronos** [comment]
Hi @Johann-S, what I called "broken" is indeed that the slide animation does not work. Can you confirm that when you visit https://getbootstrap.com/docs/4.0/components/carousel/  (which uses Bootstrap 4.0.0) you do see the slide animation?

**Comment by alecpl** [comment]
Sliding works for me with Firefox 63.0 (64-bit) on Ubuntu. I see no difference in behavior between Firefox and Chrome.

**Comment by patrickhlauke** [comment]
this relates to the `prefers-reduced-motion` media query which we added to bootstrap, and how Firefox 63 now adopts this (see https://bugzilla.mozilla.org/show_bug.cgi?id=1365045)

for windows 10 version: Firefox now looks at the "Show animations in Windows" option in windows settings. if that's set to off, it interprets this as a desire to have no animations overall and the `prefers-reduced-motion: reduce` media feature kicks in, which removes all transitions.

![capture](https://user-images.githubusercontent.com/895831/47604591-cf1f1080-d9f3-11e8-93ff-99757f51ca80.PNG)

personally, i think this is a bit of a heavy-handed approach on the part of Firefox. i hate the Win 10 animations because they're twee slow me down when working...not because i prefer no animations ever in any content. though arguably it's more the fact that - to my knowledge - it's not possible to turn off Windows' specific animations separately while keeping this global option set to on.

on macOS, Firefox looks at the accessibility option for reduced motion - this makes more sense, as this is an active choice overall to reduce motion, not just "don't show the animations in the OS".

<img width="780" alt="screenshot 2018-10-27 at 14 32 28" src="https://user-images.githubusercontent.com/895831/47604725-7d778580-d9f5-11e8-8b81-f94db068b942.png">

for Linux i have no idea where/if there's a global option somewhere for a user to signal they prefer reduced motion...

in any case, this behaves correctly (albeit surprisingly for some users).

**Comment by patrickhlauke** [comment]
@neoschronos can you confirm if you have the "reduce motion" setting enabled in macOS by any chance?

**Comment by patrickhlauke** [comment]
oh and as for macOS/Safari, same deal ... Safari looks at that "reduce motion" setting, which then sets off the `prefers-reduced-motion` media feature test.

see https://getbootstrap.com/docs/4.1/getting-started/accessibility/#reduced-motion (noting the copy there now needs updating as it's not just Safari/macOS anymore)

**Comment by patrickhlauke** [comment]
@mdo @Johann-S wonder if we should explicitly add a little note, with reference to https://getbootstrap.com/docs/4.1/getting-started/accessibility/#reduced-motion, to specific components that are visibly affected/obey the reduced motion setting to avoid this sort of confusion?

**Comment by neoschronos** [comment]
@patrickhlauke yes, the "reduce motion" setting was enabled in my settings. After disabling, slide animation "works" on Firefox 63.0 and Safari 12.0. Many thanks.

W.r.t to the "confusion"-level: it will depend on how many users have the "reduce motion" setting enabled. If there are many, honoring `prefers-reduced-motion` will results to bad user experience (I will go on a limb and say it is a big deal whether a carousel slides smoothly, or flashes the images, as is the case when the browsers honor the `prefers-reduced-motion` media query you added). Every Boootstrap 4.1.3+ owner might end up having to provide an explanation like you did with me  ... that is not optimal.

Therefore, In addition to your suggestion for a note about this at specific components, I would also suggest documenting a way for developers to disable the `prefers-reduced-motion` media query in Bootstrap (at build-time or programatically). Again, whether this is a good idea, depends on how many users have the "reduce motion" setting enabled on their accessibility settings without knowing they do.

**Comment by patrickhlauke** [comment]
> To clarify: I did not set "Reduce motion" in my accessibility settings.

it's not something that's set by default. so i'm wondering how this happened...

**Comment by Johann-S** [comment]
Yep that was me who set "Show animations in Windows" to "Off" because it slow down my computer.

Maybe we should provide a way in our carousel to force the animation ? (adding a css class or something)

**Comment by patrickhlauke** [comment]
i think having it as a configurable option for authors to enable/disable the `prefers-reduced-motion` stuff in general would be good (rather than now doing one specifically for carousel). incidentally, i'm floating the idea here about windows needing to be more specific/granular https://twitter.com/somelaniesaid/status/1056554604530425861

**Comment by XhmikosR** [comment]
Agreed, an option to toggle this should be fine for now but we might consider default this to enabled in v5.

**Comment by zaqxen** [comment]
> 
> 
> Carousel work on my FF 63 with Win10 but I don't have the slide animation 😨

if you are using bootstrap (CSS) 4.1 or higher you can fix it by adding inline style 
```
<style>
.carousel-item{
		transition: -webkit-transform .6s ease;
		transition: transform .6s ease;
		transition: transform .6s ease,-webkit-transform .6s ease;
	}
</style>
```
Had the same issue so I took the CSS animation which was working on Bootstrap 4.0

**Comment by XhmikosR** [comment]
Please read the above comments.
